src/int/mul.rs

```diff
@@ -66,7 +66,9 @@ macro_rules! mulo {
     }
 }
 
+#[cfg(not(all(feature = "c", target_arch = "x86")))]
```

Comment:
There's a comment about this in src/lib.rs. We have generic implementations of some intrinsics written in Rust but for some of those intrinsics, compiler-rt has arch-optimized implementations written in assembly. We want to use compiler-rt arch optimized impls where available; in those cases, build.rs will compile them and these cfgs remove our Rust impl from the library so we don't end with duplicate symbols.